File: dmn-core/src/test/resources/tck/cl3/expected/dmn/0004-lending/StrategyRuleOutput.java

import java.util.*;

@javax.annotation.Generated(value = {"decisionTableRuleOutput.ftl", "Strategy"})
public class StrategyRuleOutput extends com.gs.dmn.runtime.RuleOutput {
    private String strategy;

    public StrategyRuleOutput(boolean matched) {
        super(matched);
    }

    public String getStrategy() {
        return this.strategy;
    }
    public void setStrategy(String strategy) {
        this.strategy = strategy;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        StrategyRuleOutput other = (StrategyRuleOutput) o;
        if (this.getStrategy() != null ? !this.getStrategy().equals(other.getStrategy()) : other.getStrategy() != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = 0;
        result = 31 * result + (this.getStrategy() != null ? this.getStrategy().hashCode() : 0);

        return result;
    }

    public String toString() {
        StringBuilder result_ = new StringBuilder("(matched=" + isMatched());
        result_.append(String.format(", strategy='%s'", strategy));
        result_.append(")");
        return result_.toString();
    }
}

File: dmn-test-cases/proto/1.1/0004-lending/translator/expected/proto3/kotlin/dmn/StrategyRuleOutput.kt

@javax.annotation.Generated(value = ["decisionTableRuleOutput.ftl", "Strategy"])
class StrategyRuleOutput(matched: Boolean) : com.gs.dmn.runtime.RuleOutput(matched) {
    @com.fasterxml.jackson.annotation.JsonProperty("Strategy")
    var strategy: String? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as StrategyRuleOutput
        if (if (this.strategy != null) this.strategy != other.strategy else other.strategy != null) return false

        return true
    }

    override fun hashCode(): Int {
        var result = 0
        result = 31 * result + (if (this.strategy != null) this.strategy.hashCode() else 0)

        return result
    }

    override fun toString(): String {
        val result_ = StringBuilder("(matched=" + isMatched())
        result_.append(String.format(", strategy='%s'", strategy))
        result_.append(")")
        return result_.toString()
    }
}
